Mage.Sets/src/mage/cards/x/XanatharGuildKingpin.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    private XanatharGuildKingpinRuleModifyingEffect(final mage.cards.x.XanatharGuildKingpinRuleModifyingEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public mage.cards.x.XanatharGuildKingpinRuleModifyingEffect copy() {
+        return new mage.cards.x.XanatharGuildKingpinRuleModifyingEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public String getInfoMessage(Ability source, GameEvent event, Game game) {
+        Player targetPlayer = game.getPlayer(getTargetPointer().getFirst(game, source));
+        MageObject mageObject = game.getObject(source.getSourceId());
+        if (targetPlayer != null && mageObject != null) {
+            return "This turn you can't cast spells" +
+                    " (" + mageObject.getLogName() + ')';
+        }
+        return null;
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.CAST_SPELL;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        return event.getPlayerId().equals(getTargetPointer().getFirst(game, source));
+    }
+}
+
+class XanatharGuildKingpinLookAtTopCardOfTargetPlayerLibraryAnyTimeEffect extends ContinuousEffectImpl {
+
+    public XanatharGuildKingpinLookAtTopCardOfTargetPlayerLibraryAnyTimeEffect() {
+        super(Duration.EndOfTurn, Layer.PlayerEffects, SubLayer.NA, Outcome.Benefit);
+        staticText = "Until end of turn, you may look at the top card of target player's library any time.";
+    }
+
+    private XanatharGuildKingpinLookAtTopCardOfTargetPlayerLibraryAnyTimeEffect(final mage.cards.x.XanatharGuildKingpinLookAtTopCardOfTargetPlayerLibraryAnyTimeEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (game.inCheckPlayableState()) { // Ignored - see https://github.com/magefree/mage/issues/6994
+            return false;
+        }
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        Player targetPlayer = game.getPlayer(source.getFirstTarget());
+        if (targetPlayer == null) {
+            return false;
+        }
+
+        Card topCard = targetPlayer.getLibrary().getFromTop(game);
+        if (topCard == null) {
+            return false;
+        }
+
+        if (!canLookAtNextTopLibraryCard(game)) {
+            return false;
+        }
+        controller.lookAtCards("Top card of " + targetPlayer.getName() + "'s library", topCard, game);
+
+        ContinuousEffect effect = new XanatharGuildKingpinSpendManaAsThoughEffect(controller.getId());
+        effect.setTargetPointer(new FixedTarget(topCard.getId(), game));
+        game.addEffect(effect, source);
```

Comment:
You can't generate ContinuousEffect in another ContinuousEffect (ContinuousEffect applies on each game cycle, e.g. dozens per step). It must be one time action (e.g. you must add look and play effects in same place like one shot effect -- so you will have only 2 effects per turn).

Reply:
Example 1:
![shot_210725_125356](https://user-images.githubusercontent.com/8344157/126893767-c9d56532-f821-4760-b211-c7dc428c9921.png)

Example 2:
![shot_210725_130819](https://user-images.githubusercontent.com/8344157/126893879-9db51297-56da-4cb2-b2fd-c029532d25ab.png)

Reply:
Yeah I tried something like that at first:
addAbility(LookAtTheTopCardOfTargetPlayerAnyTimeEffect)
addAbility(PlayTheTopCardOfTargetPlayerLibraryAnyTimeEffect)
addAbility(PlayTheTopCardOfTargetPlayerLibraryAsThoughAnyManaEffect)

The problem that I was facing was that the last ContinuousEffect did not apply after the card went onto the stack (since it was not on the top of target players library anymore).

I saw that for [Thief of Sanity](https://scryfall.com/card/grn/205/thief-of-sanity) there was a similiar problem and decided to do it like it has been done there.
But ye I see.  I guess doing it like this inside a ContinuousEffect as I did will then spawn a new instance of the XanatharGuildKingpinSpendManaAsThoughEffect on every check right? 😕

Reply:
@JayDi85 Okay I think I found a nice way to make this whole thing much cleaner :D
Added a new commit.

I am not sure though if the wording of the effect should be configured differently.

Reply:
Ok damn nevermind I realised that `makeCardPlayable` just adds a ContinuousEffect as well

Reply:
Okay I found a way of initializing the SpendManaAsAnyColor with a OneShotEffect.
Added a commit for that.